### ISSUE TITLE
Ambiguous field identifier when parsing unions with unqualified columns and parseWithMetaLookups

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10968]
Given this schema:

```sql
create table t1 (a int);
create table t2 (a int);
```

Parsing this valid SQL:

```sql
select a
from t1
union all
select a
from t2
```

Yields:

```
Ambiguous field identifier: [2:1] select a
[*]from t1
union all
select a
from t2
```

----
Related: https://github.com/jOOQ/jOOQ/issues/10699

**Comment by lukaseder** [comment]
Not a 3.15 regression, the issue is also present in 3.14.4

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.4 (#10969)
